ci-operator/step-registry/azure/cluster/public-to-private/azure-cluster-public-to-private-commands.sh

```diff
@@ -0,0 +1,180 @@
+api_public_ip_id=$(az network lb frontend-ip show --name ${frontend_ip} --lb-name ${INFRA_ID} -g ${RESOURCE_GROUP} --query 'publicIPAddress.id' -otsv)
+
+echo "Restricting the API server to private - delete inbound rule api-v4 from external lb"
+run_command "az network lb rule delete -n api-v4 --lb-name ${INFRA_ID} -g ${RESOURCE_GROUP}"
+
+echo "Restricting the API server to private - delete associated frontend public ip from external lb"
+run_command "az network lb frontend-ip delete --name ${frontend_ip} --lb-name ${INFRA_ID} -g ${RESOURCE_GROUP}"
+
+echo "Restricting the API server to private - delete frontend public IP associated with rule api-v4"
+run_command "az network public-ip delete --ids ${api_public_ip_id}"
+
+#api-v4 inbound rule is deleted from external lb, need to export proxy to access the cluster
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+else
+  echo "No proxy-conf found, exit now"
+  exit 1
+fi
+
+echo "Restricting the API server to private - delete api dns record from public dns zone"
+run_command "az network dns record-set cname delete --name api.${CLUSTER_NAME} --zone-name ${BASE_DOMAIN} -g ${BASE_DOMAIN_RESOURCE_GROUP}"
+
+# Configure a private storage endpoint on Azure
+echo "Configure a private storage endpoint on Azure by enabling the image registry operator to discover vnet and subnet names"
+vnet_resource_group=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.networkResourceGroupName')
+vnet_name=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.virtualNetwork')
+vnet_subnet_name=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.controlPlaneSubnet')
+if [[ -n "${vnet_resource_group}" ]]; then
+    run_command "oc patch config.image/cluster -p '{\"spec\":{\"storage\":{\"azure\":{\"networkAccess\":{\"type\": \"Internal\",\"internal\":{\"networkResourceGroupName\":\"${vnet_resource_group}\"}}}}}}' --type=merge"
+else
+    vnet_resource_group=${RESOURCE_GROUP}
+    vnet_name="${INFRA_ID}-vnet"
+    vnet_subnet_name="${INFRA_ID}-master-subnet"
+    run_command "oc patch config.image/cluster -p '{\"spec\":{\"azure\":{\"networkAccess\":{\"type\": \"Internal\"}}}}' --type=merge"
+fi
+
+# FIXME: add this when https://issues.redhat.com/browse/OCPBUGS-30196 is fixed
+# Optional: Disabling redirect when using a private storage endpoint on Azure
+
+echo "Waiting for image registry become Available..."
+run_command "oc wait --for=condition=Available --for condition=Progressing=False --for condition=Degraded=False co/image-registry --timeout=1200s"
+
+# Check image-registry config
+image_registry_sa_name=$(az storage account list -g ${RESOURCE_GROUP} --query '[].name' -otsv | grep imageregistry)
+image_registry_private_endpoint=$(az network private-endpoint list -g ${RESOURCE_GROUP} --query '[].name' -otsv)
```

Comment:
I saw common IPI private cluster does not have the private endpoint, is it not empty here in public-to-private case?

Reply:
private endpoint is the newly created by image registry when registry type is changed from External to Internal. In common IPI private cluster, the default type of registry is External, no private endpoint to be created.